src/mlpack/methods/ann/layer/nearest_interpolation.hpp

```diff
@@ -72,52 +76,36 @@ class NearestInterpolation
    * @param gradient The computed backward gradient.
    * @param output The resulting down-sampled output.
    */
-  template<typename eT>
-  void Backward(const arma::Mat<eT>& /*input*/,
-                const arma::Mat<eT>& gradient,
-                arma::Mat<eT>& output);
-
-  //! Get the output parameter.
-  OutputDataType const& OutputParameter() const { return outputParameter; }
-  //! Modify the output parameter.
-  OutputDataType& OutputParameter() { return outputParameter; }
-
-  //! Get the delta.
-  OutputDataType const& Delta() const { return delta; }
-  //! Modify the delta.
-  OutputDataType& Delta() { return delta; }
+  void Backward(const MatType& /*input*/,
+                const MatType& gradient,
+                MatType& output);
 
+  void ComputeOutputDimensions();
   //! Get the row size of the input.
   size_t const& InRowSize() const { return inRowSize; }
   //! Modify the row size of the input.
   size_t& InRowSize() { return inRowSize; }
-
+  
   //! Get the column size of the input.
   size_t const& InColSize() const { return inColSize; }
   //! Modify the column size of the input.
   size_t& InColSize() { return inColSize; }
-
+  
```

Comment:
you are adding trailing whitespaces at the start of these lines, you can detect these I think with vim easily